File: templates/backend/django/levels/level_3/files/backend/app/repositories/todo_repository.py
class TodoRepository:
    def __init__(self) -> None:
        self._todos: dict[int, dict[str, object]] = {}
        self._next_id = 1

    def list(self) -> list[dict[str, object]]:
        return list(self._todos.values())

    def create(self, title: str) -> dict[str, object]:
        todo = {"id": self._next_id, "title": title, "completed": False}
        self._todos[self._next_id] = todo
        self._next_id += 1
        return todo

File: templates/backend/nodejs/levels/level_3/files/backend/src/repositories/todoRepository.js
class TodoRepository {
  constructor() {
    this.todos = new Map();
    this.nextId = 1;
  }

  list() {
    return Array.from(this.todos.values());
  }

  create(title) {
    const todo = { id: this.nextId, title, completed: false };
    this.todos.set(this.nextId, todo);
    this.nextId += 1;
    return todo;
  }
}

module.exports = { TodoRepository };
